dmutils/email/dm_notify.py

```diff
@@ -9,6 +13,13 @@
 from dmutils.timing import logged_duration_for_external_request as log_external_request
 
 
+class DMNotifyEmail(NamedTuple):
+    to_email_address: str
+    template_name_or_id: str
+    personalisation: Optional[Dict[str, str]] = None
+    reference: Optional[str] = None
```

Comment:
Nice.

Though, why is `reference` optional?

Reply:
Ah, that's a holdover from a different approach I tried and gave up on, I will fix that, thanks.